### PR TITLE
Add Dockerfile for Noetic

### PR DESCRIPTION
**Pull request by Serafadam**
As in the title

### CODE REVIEW

**Review by themarpe — CHANGES_REQUESTED** [review]
Tag `depthai` is already taken: https://hub.docker.com/r/luxonis/depthai - used for `depthai` repository tooling, demo, etc...

Also, I'd suggest renaming to just `Dockerfile` to not require specifying it manually?

File: README.md [hunk 1]
```diff
@@ -18,6 +18,30 @@ sudo udevadm control --reload-rules && sudo udevadm trigger
 Install depthai-ros. (Available for Noetic, foxy, galactic and humble)
 `sudo apt install ros-<distro>-depthai-ros`
 
+## Docker
+You can additionally build and run docker images on your local machine. To do that, add USB rules as in above step, clone the repository and inside it run (it matters on which branch you are on):
+```
+docker build --build-arg USE_RVIZ=1 -f ./desktop.dockerfile -t depthai_ros .
```
Comment (themarpe) [inline README.md]:
```suggestion
docker build --build-arg USE_RVIZ=1 -t depthai-ros .
```

**Review by themarpe — CHANGES_REQUESTED** [review]
Tag should be `depthai-ros` as is with others: https://hub.docker.com/search?q=depthai

Also should prebuilt docker image have RVIZ enabled? And which branch does it target? Otherwise we'll have to make it per branch instead of just `latest` in CI - if those differ by branch (eg underlying base perhaps?)

File: README.md (hunk 1 above)
Comment (Serafadam) [reply README.md]: I've changed the tag name to reflect on the branch name, as when it comes to installing rviz, current launch files run it by default, so maybe we should keep it?

Comment (themarpe) [reply README.md]: I was thinking more in the lines of adding it as a "tag" on the depthai-ros name. 
So should at the end be: `luxonis/depthai-ros:${BRANCH_NAME}`

and then there is not latest (or latest is whatever the latest ROS OS base is)

**Comment by themarpe** [comment]
Also, is it necessary to have separate branches per ROS version (instead of one that can "dynamiclly" handle all versions)?
Eitherway, we should make it such that these contents are the same, and just the branch name is what is used to actually make the difference. Then updating branches is easy (without sneaking in any other branches details)

**Comment by Serafadam** [comment]
> Also, is it necessary to have separate branches per ROS version (instead of one that can "dynamiclly" handle all versions)? Eitherway, we should make it such that these contents are the same, and just the branch name is what is used to actually make the difference. Then updating branches is easy (without sneaking in any other branches details)

I'm keeping this up with convention that other ROS projects follow, since main development will happen on ROS Rolling/Humble/Noetic and other branches serve as a clear starting point if we want to backport the changes. I totally agree on branch names being variables

File: README.md (hunk 1 above)
Comment (themarpe) [reply README.md]: > to installing rviz, current launch files run it by default, so maybe we should keep it?

I mean, you expose it as a parameter when building locally, but you do NOT build with it, in CI. 
I assume we'd want rather more complex prebuilt docker images than not, such that more things can be done out of the box.

TLDR add the build arg `USE_RVIZ` to CI when building the images as well? Or is it ON by default?

Comment (Serafadam) [reply README.md]: Updated flags, modified CI file a little bit and reformatted the code, now it all lights up green :slightly_smiling_face:

**Review by themarpe — COMMENTED** [review]
The push should happen on tag likely (otherwise commit?)

**Review by themarpe — APPROVED** [review]
LGTM

**Comment by Serafadam** [comment]
Needs one quick fix tommorow and ready to merge
